webpay/pay/views.py

```diff
@@ -155,14 +155,17 @@ def configure_transaction(request, data=None):
         log.error('Configuring transaction failed: {er}'.format(er=error_code))
         return system_error(request, code=error_code)
     else:
-        sim = (request.session['notes']['pay_request']['request']['simulate']
-               if is_simulation else None)
+        pay_req = request.session['notes'].get('pay_request')
+        payment_required = (
+            pay_req['request']['pricePoint'] != '0' if pay_req else True)
+        sim = pay_req['request']['simulate'] if is_simulation else None
```

Comment:
this will blow up if pay_req is falsey so I think you can assume it will always be there.